crates/ide/src/children_module.rs

```diff
@@ -0,0 +1,150 @@
+use hir::Semantics;
+use ide_db::{FilePosition, RootDatabase};
+use syntax::{
+    algo::find_node_at_offset,
+    ast::{self, AstNode},
+};
+
+use crate::NavigationTarget;
+
+// Feature: Children Module
+//
+// Navigates to the children modules of the current module.
+//
+// | Editor  | Action Name |
+// |---------|-------------|
+// | VS Code | **rust-analyzer: Locate children module** |
+//
+// ![Children Module]()
+
+/// This returns `Vec` because a module may be included from several places.
+pub(crate) fn children_module(db: &RootDatabase, position: FilePosition) -> Vec<NavigationTarget> {
+    let sema = Semantics::new(db);
+    let source_file = sema.parse_guess_edition(position.file_id);
+    // First go to the parent module which contains the cursor
+    let mut module = find_node_at_offset::<ast::Module>(source_file.syntax(), position.offset);
+
+    // If cursor is literally on `mod foo`, go to the grandpa.
+    if let Some(m) = &module {
+        if !m
+            .item_list()
+            .is_some_and(|it| it.syntax().text_range().contains_inclusive(position.offset))
+        {
+            cov_mark::hit!(test_resolve_parent_module_on_module_decl);
+            module = m.syntax().ancestors().skip(1).find_map(ast::Module::cast);
+        }
+    }
```

Comment:
I think we want to show the children of the `mod foo` in this case, so this here should not be necessary